docs/csharp/programming-guide/concepts/async/walkthrough-accessing-the-web-by-using-async-and-await.md

```diff
@@ -1,379 +1,369 @@
+    Copy the following code, and paste it above the `OnStartButtonClick` event handler in *MainWindow.xaml.cs*:
 
-    - `DisplayResults`, which displays  the number of bytes in the byte array for each URL.
+    ```csharp
+    private readonly IEnumerable<string> _urlList = new string[]
+    {
+        "https://docs.microsoft.com",
+        "https://docs.microsoft.com/azure",
+        "https://docs.microsoft.com/powershell",
+        "https://docs.microsoft.com/dotnet",
+        "https://docs.microsoft.com/aspnet/core",
+        "https://docs.microsoft.com/windows"
+    };
+    ```
 
-    Copy the following four methods, and then paste them under the `startButton_Click` event handler in MainWindow.xaml.cs:
+    Copy the following methods, and paste them below the `OnStartButtonClick` event handler in *MainWindow.xaml.cs*:
 
     ```csharp
     private void SumPageSizes()
     {
-        // Make a list of web addresses.
-        List<string> urlList = SetUpURLList();
+        var stopwatch = Stopwatch.StartNew();
 
-        var total = 0;
-        foreach (var url in urlList)
-        {
-            // GetURLContents returns the contents of url as a byte array.
-            byte[] urlContents = GetURLContents(url);
-
-            DisplayResults(url, urlContents);
+        int total = _urlList.Select(url => ProcessUrl(url)).Sum();
 
-            // Update the total.
-            total += urlContents.Length;
-        }
-
-        // Display the total count for all of the web addresses.
-        resultsTextBox.Text += $"\r\n\r\nTotal bytes returned:  {total}\r\n";
+        stopwatch.Stop();
+        _resultsTextBox.Text += $"\nTotal bytes returned:  {total:#,#}";
+        _resultsTextBox.Text += $"\nElapsed time:          {stopwatch.Elapsed}\n";
     }
 
-    private List<string> SetUpURLList()
+    private int ProcessUrl(string url)
     {
-        var urls = new List<string>
-        {
-            "https://msdn.microsoft.com/library/windows/apps/br211380.aspx",
-            "https://msdn.microsoft.com",
-            "https://msdn.microsoft.com/library/hh290136.aspx",
-            "https://msdn.microsoft.com/library/ee256749.aspx",
-            "https://msdn.microsoft.com/library/hh290138.aspx",
-            "https://msdn.microsoft.com/library/hh290140.aspx",
-            "https://msdn.microsoft.com/library/dd470362.aspx",
-            "https://msdn.microsoft.com/library/aa578028.aspx",
-            "https://msdn.microsoft.com/library/ms404677.aspx",
-            "https://msdn.microsoft.com/library/ff730837.aspx"
-        };
-        return urls;
-    }
-
-    private byte[] GetURLContents(string url)
-    {
-        // The downloaded resource ends up in the variable named content.
-        var content = new MemoryStream();
-
-        // Initialize an HttpWebRequest for the current URL.
+        using var memoryStream = new MemoryStream();
         var webReq = (HttpWebRequest)WebRequest.Create(url);
```

Comment:
The API reference for HttpWebRequest says:

> We don't recommend that you use HttpWebRequest for new development. Instead, use the System.Net.Http.HttpClient class.

Do you want to me to open an issue to address this separately in another PR or address it in this PR (and the dotnet/samples PR too)?

Reply:
Since we're making this many updates, I'd add it to this PR

/cc @IEvangelist

Reply:
The tutorial explicitly covers updating from `HttpWebRequest` to `HttpClient`. I think we should leave it, thoughts @BillWagner?

Reply:
Oh, My bad. I see that the final code uses HttpClient. It should be okay. But probably the other sample that's explained to be extended here may need to get updated to HttpClient? In that case, it may be another task.

Reply:
What other examples are you referring to? The effort would more than likely include articles that refer to them.